kratos/includes/table.h

```diff
@@ -603,24 +615,25 @@ class Table<double, double>
         return mData[size-1].second[0];
     }
 
-    result_type& Interpolate(argument_type const& X, argument_type const& X1, result_type const& Y1, argument_type const& X2, result_type const& Y2, result_type& Result) const
+    result_type Interpolate(argument_type const& X, argument_type const& X1, result_type const& Y1, argument_type const& X2, result_type const& Y2) const
     {
-        const double epsilon = 1e-12;
-
-        double dx = X2 - X1;
-        result_type dy = Y2 - Y1;
+        // const double epsilon = 1e-12;
+        return Y1 + (X2 - X1 > 1.0e-12) * (Y2 - Y1)*(X - X1)/(X2 - X1);
```

Comment:
Any especial reason to take out the epsilon? You may make it constexp and leave it for better readability